diff --git a/chrome/browser/search_engines/template_url_service_factory.cc b/chrome/browser/search_engines/template_url_service_factory.cc
index b3f207c4d728b68fa725654e916f0d8d1e3b0c49..58d55a22ca188cb7a6c19cb49654c76be9094ffd 100644
--- a/chrome/browser/search_engines/template_url_service_factory.cc
+++ b/chrome/browser/search_engines/template_url_service_factory.cc
@@ -82,6 +82,8 @@ void TemplateURLServiceFactory::RegisterProfilePrefs(
 
 content::BrowserContext* TemplateURLServiceFactory::GetBrowserContextToUse(
     content::BrowserContext* context) const {
+  // To make different service for normal and incognito profile.
+  return chrome::GetBrowserContextOwnInstanceInIncognito(context);
   return chrome::GetBrowserContextRedirectedInIncognito(context);
 }
 
